src/core/meshcop/tcat_agent.cpp

```diff
@@ -0,0 +1,519 @@
+    {
+    case TlvType::kSetActiveOperationalDataset:
+        error = HandleActiveDataset(aIncommingMessage, offset, tlv.GetLength());
+        break;
+
+    default:
+        error = kErrorParse;
+    }
+
+exit:
+    return error;
+}
+
+Error TcatAgent::HandleGeneral(ot::Message &aIncommingMessage,
+                               ot::Message &aOutgoingMessage,
+                               uint8_t      aTlvType,
+                               uint32_t     aOffset,
+                               uint32_t     aLength,
+                               bool        &aHasResponse)
+{
+    Error       error;
+    CommandType command;
+
+    OT_UNUSED_VARIABLE(aOutgoingMessage);
+    OT_UNUSED_VARIABLE(aHasResponse);
+
+    switch (aTlvType)
+    {
+    case TlvType::kCommand:
+        VerifyOrExit(aLength == sizeof(command), error = kErrorParse);
+        SuccessOrExit(error = aIncommingMessage.Read(aOffset, &command, sizeof(command)));
+        error = HandleCommand(command, aOutgoingMessage);
+        break;
+
+    case TlvType::kDisconnect:
+        error = kErrorAbort;
+        break;
+
+    default:
+        error = kErrorInvalidCommand;
+    }
+
+exit:
+    return error;
+}
+
+Error TcatAgent::HandleCommissioning(ot::Message &aIncommingMessage,
+                                     ot::Message &aOutgoingMessage,
+                                     uint8_t      aTlvType,
+                                     uint32_t     aOffset,
+                                     uint32_t     aLength,
+                                     bool        &aHasResponse)
+{
+    Error error;
+
+    OT_UNUSED_VARIABLE(aOutgoingMessage);
+    OT_UNUSED_VARIABLE(aHasResponse);
+
+    if (!IsCommandClassAuthorized(mCommissionerAuthorizationField.mCommissioningFlags,
+                                  mDeviceAuthorizationField.mCommissioningFlags))
+    {
+        error = kErrorRejected;
+        ExitNow();
+    }
+
+    switch (aTlvType)
+    {
+    case TlvType::kSetActiveOperationalDataset:
+        error = HandleActiveDataset(aIncommingMessage, aOffset, aLength);
+        break;
+
+    default:
+        error = kErrorInvalidCommand;
+    }
+
+exit:
+    return error;
+}
+
+Error TcatAgent::HandleExtraction(ot::Message &aIncommingMessage,
+                                  ot::Message &aOutgoingMessage,
+                                  uint8_t      aTlvType,
+                                  uint32_t     aOffset,
+                                  uint32_t     aLength,
+                                  bool        &aHasResponse)
+{
+    Error error;
+
+    OT_UNUSED_VARIABLE(aIncommingMessage);
+    OT_UNUSED_VARIABLE(aOutgoingMessage);
+    OT_UNUSED_VARIABLE(aOffset);
+    OT_UNUSED_VARIABLE(aLength);
+    OT_UNUSED_VARIABLE(aHasResponse);
+
+    switch (aTlvType)
+    {
+    default:
+        error = kErrorInvalidCommand;
+    }
+
+    // exit:
+    return error;
+}
+
+Error TcatAgent::HandleDecommissioning(ot::Message &aIncommingMessage,
```

Comment:
`HandleExtraction()` and `HandleDecommissioning()` do nothing today.
Should we remove them and add when we support this?

Reply:
removed.